zebra-state/src/service/non_finalized_state.rs

```diff
@@ -456,7 +458,7 @@ impl NonFinalizedState {
 
         let mut modified_chain = Arc::unwrap_or_clone(chain_result);
         for block in invalidated_blocks {
-            modified_chain = modified_chain.push(block).map_err(Box::new)?;
+            modified_chain = modified_chain.push(block)?;
```

Comment:
The `ReconsiderError::ValidationError` variant can be removed since pushing a previously existing but invalidated block onto a chain can be expected to succeed.

This method will need an `#[allow(unwrap_in_result)]` attribute to silence a lint.

```suggestion
            modified_chain = modified_chain
                .push(block)
                .expect("previously invalidated block should be valid for chain");
```